### PR TITLE
feat(@schematics/angular): fetch latest node package version

### PR DESCRIPTION
**Pull request by schuchard**
Add method to get a npm packages latest registry version and provide a default value of "latest" if any errors occur. 

This is desired when creating a schematic that relies external packages with no specific package version.

Example usage when adding a dependency using new `getLatestNodeVersion`:

```typescript
of('jest', 'jest-preset-angular').pipe(
  concatMap((packageName: string) => getLatestNodeVersion(packageName)),
  map(({ name, version }: NodePackage) => {

    addPackageJsonDependency(tree, {
      type: NodeDependencyType.Dev,
      name,
      version,
    });
...
```

### CODE REVIEW

**Comment by Brocco** [comment]
Will review once CI is green, please address the current issues, thanks!

**Comment by schuchard** [comment]
Sorry about that @Brocco, back in the green. First PR in this repo so open to suggestions, thanks.

File: packages/schematics/angular/package.json [hunk 1]
```diff
@@ -14,6 +14,8 @@
   "dependencies": {
     "@angular-devkit/core": "0.0.0",
     "@angular-devkit/schematics": "0.0.0",
+    "@types/nock": "^9.1.3",
```
[the remark this answers is not in the document]
Comment (schuchard) [reply packages/schematics/angular/package.json]: Originally these deps were missing in this package.json and it caused a CI lint failure. I've pushed another commit with them removed for you look at. If the nock packages are missing from this package.json, the following lint error occurs.

```shell
angular-cli/packages/schematics/angular/utility/dependencies_spec.ts
Module 'nock' is not listed as dependency in package.json (no-implicit-dependencies)
   8 | import { EmptyTree } from '@angular-devkit/schematics';
   9 | import { UnitTestTree } from '@angular-devkit/schematics/testing';
> 10 | import * as nock from 'nock';
     |                      ^
  11 | import {
  12 |   NodeDependency,
  13 |   NodeDependencyType,
```

The same error occurs if I put the nock deps in `devDependencies` and with a `require` usage in the spec. Any idea's on how to fix this?

**Comment by schuchard** [comment]
Looks like my response was collapsed in the package.json comment. Re-posting here just in case it was missed. 

Originally these deps were missing in this package.json and it caused a CI lint failure. I've pushed another commit with them removed for you look at. If the nock packages are missing from this package.json, the following lint error occurs.

```shell
angular-cli/packages/schematics/angular/utility/dependencies_spec.ts
Module 'nock' is not listed as dependency in package.json (no-implicit-dependencies)
   8 | import { EmptyTree } from '@angular-devkit/schematics';
   9 | import { UnitTestTree } from '@angular-devkit/schematics/testing';
> 10 | import * as nock from 'nock';
     |                      ^
  11 | import {
  12 |   NodeDependency,
  13 |   NodeDependencyType,
```

The same error occurs if I put the nock deps in `devDependencies` and with a `require` usage in the spec. Any idea's on how to fix this?

**Comment by hansl** [comment]
I don't think this code is used anymore. Are there any reasons? This seems like something we would like in the future. I'll leave it open for now, and mark it as needing discussion.

**Comment by hansl** [comment]
I'm going to close this with a story in our backlog for improving on using floating versions of Angular on `ng new` (and supporting caches etc). This will be covered in the future.

**Comment by schuchard** [comment]
I ran into this missing feature on a few schematics (https://github.com/briebug/jest-schematic and https://github.com/schuchard/prettier-schematic). I'm interested in contributing to that feature when the story's available.

**Comment by schuchard** [comment]
I can agree with that. I would definitely be in favor of seeing those utilities as a consumable library. There's a lot of great utilities in the Angular schematics repo!

**Pull request closed without merging**